packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -348,24 +355,40 @@ class IconButton extends StatelessWidget {
       );
     }
 
+    final MaterialStateProperty<Color?> overlayColor =
+      _IconButtonDefaultOverlay(
+        defaultColor: Colors.red,
+        splashColor: splashColor ?? theme.splashColor,
+        focusColor: focusColor ?? theme.focusColor,
+        hoverColor: hoverColor ?? theme.hoverColor,
+      );
+    final MaterialStateProperty<MouseCursor?> cursor = mouseCursor != null ? MaterialStateProperty.all<MouseCursor?>(mouseCursor) :
+      _IconButtonDefaultMouseCursor(SystemMouseCursors.click, SystemMouseCursors.forbidden);
+
     return Semantics(
       button: true,
       enabled: onPressed != null,
-      child: InkResponse(
+      child: TextButton(
```

Comment:
As you've noted. what's different about TextButton (and the other ButtonStyle buttons) and IconButton is that the IconButton's splash radius is bigger than the button itself. It might be possible to extend ButtonStyleButton to handle this special case by conditionally adding an additional transparent Material layer that was big enough to accommodate the splash. That would likely break tests that assume that the Material above a button is where the button's background, shape, etc are to be found. Perhaps a better way to handle this would be to improve the IconButton's documentation with an example using transparent Material. The current documentation is obscure: "picture printed on a Material widget" doesn't clarify things very well since we don't "print", and it doesn't explain what the implications of rendering the button on whatever the underlying Material happens to be.